src/common/actions/auth-store.js

```diff
@@ -270,7 +272,13 @@ async function setShortNamespace(root, discharge, userName) {
     }
     return data;
   } else {
-    const json = await response.json();
+    const text = await response.text();
```

Comment:
OK, the specific case that the previous code was handling was a 204 response from PATCH with an empty response body (as befits 204).  That's HTTPishly correct since PATCH has no content to send in its response, but we have to take care to cope with it.

However, now that this code is on the client side, the 204 case is already handled by the other branch of this `if` which entirely ignores the response body, so there's no need to handle the empty-body case specially.

I would suggest:

 * go back to the previous `const json = await response.json();` code here
 * add an XXX comment somewhere with a bug reference reminding us to do something about the general case of better handling of non-JSON error responses; I agree at this point that that doesn't need to be done in this branch, but let's not forget about it